infra/modules/monitoring/variables.tf

```diff
@@ -7,3 +7,14 @@ variable "load_balancer_arn_suffix" {
   type        = string
   description = "The ARN suffix for use with CloudWatch Metrics."
 }
+
+variable "email_alerts" {
+  type        = set(string)
+  default     = []
+  description = "List of emails for aler subscription"
+
+}
+
+variable "ssm_secret" {
```

Comment:
missing type definition for variable

Reply:
wasn't sure about the type at the moment, but it looks like a map.

Reply:
@lorenyu I'm stuck trying to write this variable type.
It is tuple of maps with specific list of attributes. However I cannot just print it into output and so I could check structure, because it is sensitive and terraform does not want print it.
even with nonsensetive function I'm getting error:
`#output "data_value" {
#  value     = nonsensitive(data.aws_ssm_parameter.incident_management_service_integration_url)
#}
`
result:
`╷
│ Error: Invalid function argument
│
│   on outputs.tf line 15, in output "data_value":
│   15:   value     = nonsensitive(data.aws_ssm_parameter.incident_management_service_integration_url)
│     ├────────────────
│     │ data.aws_ssm_parameter.incident_management_service_integration_url is tuple with 1 element
│
│ Invalid value for "value" parameter: the given value is not sensitive, so this call is redundant.
╵
make: *** [infra-update-app-service] Error 1`

I tried to goole what data type this is, but couldn't find it either.

Reply:
you're thinking about it backwards – don't ask what type is being passed in by `app/service/`, think about what type the `modules/monitoring` module _should_ be accepting. `variables.tf` for terraform modules is like function parameters for a function – when you're in the _design_ process, you want to ask yourself, what does this module/function need to do its job?

Reply:
made it less complicated. I passed actual value to the module instead of entire object.